### PR TITLE
Add RPC subscription api for rooted slots

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
The pre-existing slot subscription RPC API is not sufficient to track rooted slots. The explorer should be able to display the root progress of a cluster.

#### Summary of Changes
Add new subscription API for tracking new root slots

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
(needs a doc update for the new subscribe method please)

**Comment by jstarry** [comment]
ah right, on it

**Comment by jstarry** [comment]
Fixed!

File: core/src/rpc_subscriptions.rs [hunk 1]
```diff
@@ -576,15 +609,17 @@ pub(crate) mod tests {
...
         let (inner_sender, inner_receiver) = channel();
         let mut rt = Runtime::new().unwrap();
         rt.spawn(futures::lazy(|| {
             let recv_timeout = receiver
                 .into_future()
                 .timeout(Duration::from_millis(RECEIVE_DELAY_MILLIS))
                 .map(move |result| match result {
-                    (Some(value), _) => inner_sender.send(value).expect("send error"),
+                    (Some(value), receiver) => {
+                        inner_sender.send((value, receiver)).expect("send error")
```
Comment (CriesofCarrots) [inline core/src/rpc_subscriptions.rs]: ❗️ This is way easier than what I was trying to do with this function last week. Awesome!

Comment (jstarry) [reply core/src/rpc_subscriptions.rs]: Yes! Such a relief when I realized this was possible haha

**Pull request merged**